src/app/core/services/color-generator.service.ts

```diff
@@ -0,0 +1,39 @@
+import {Injectable} from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ColorGeneratorService {
+
+  constructor() {
+  }
+
+  getColor(amount = 1): string | string[] {
```

Comment:
Type the amount and don't make a default 'amount' and always return a string array. The chart library always needs arrays